tests/modes.spec.ts

```diff
@@ -1,19 +1,80 @@
+async function clickMap(page, x, y) {
   await page.getByRole("region", { name: "Map" }).click({
-    position: {
-      x: 500,
-      y: 500,
-    },
+    position: { x, y },
   });
+}
+
+test("clearing all while a feature is open works", async ({ page }) => {
+  await page.goto("/scheme.html?authority=Adur");
+  await page.getByRole("button", { name: "New point" }).click();
+  await clickMap(page, 500, 500);
   page.on("dialog", (dialog) => dialog.accept());
   await page.getByRole("button", { name: "Clear all" }).click();
   // Make sure AttributeMode resets to nothing being selected
   await expect(
     page.getByText("Click an object to fill out its attributes")
   ).toBeVisible();
 });
+
+test("creating a new point opens a form", async ({ page }) => {
+  await page.goto("/scheme.html?authority=Adur");
+  await page.getByRole("button", { name: "New point" }).click();
+  await clickMap(page, 500, 500);
+
+  await page.getByRole("button", { name: "1) Untitled point" }).isVisible();
+  await page.getByLabel("Description:").click();
+});
+
+test("creating a new freehand polygon opens a form", async ({ page }) => {
+  await page.goto("/scheme.html?authority=Adur");
+  await page.getByRole("button", { name: "New polygon (freehand)" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 600, 500);
+  await clickMap(page, 600, 600);
+  await page.getByRole("button", { name: "Finish" }).click();
+
+  await page.getByRole("button", { name: "1) Untitled area" }).isVisible();
+  await page.getByLabel("Description:").click();
+});
+
+test("creating a new snapped polygon opens a form", async ({ page }) => {
+  await page.goto("/scheme.html?authority=Adur");
+  await page.getByRole("button", { name: "New polygon (snapped)" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 600, 500);
+  await clickMap(page, 600, 600);
+  await page.getByRole("button", { name: "Finish" }).click();
+
+  await page.getByRole("button", { name: "1) Untitled area" }).isVisible();
+  await page.getByLabel("Description:").click();
+});
+
+test("creating a new route opens a form", async ({ page }) => {
+  await page.goto("/scheme.html?authority=Adur");
+  await page.getByRole("button", { name: "New route" }).click();
+  await clickMap(page, 500, 500);
+  await clickMap(page, 600, 500);
+  await page.getByRole("button", { name: "Finish" }).click();
+
+  await page.getByRole("button", { name: "1) Untitled route" }).isVisible();
+  await page.getByLabel("Description:").click();
+});
+
+test("other tools work when route tool doesn't load", async ({ page }) => {
+  await page.route("https://atip.uk/route-snappers-dev/Adur.bin.gz", (route) =>
+    route.fulfill({
+      status: 404,
+    })
+  );
+  await page.goto("/scheme.html?authority=Adur");
+
+  await page.getByRole("button", { name: "New route" }).click();
+  await expect(page.getByText("Failed to load")).toBeVisible();
+
+  // Other tools should still work
```

Comment:
@robinlovelace-ate, actually the point tool still works when route tool fails. Editing geometry might be broken though. I can't tell from your screenshot what happened exactly, but I'll keep playing with it and add tests to make sure other tools still work

Reply:
Will test...

Reply:
I'm pretty sure I could not add a point or anything to download as a GeoJSON in the accidental test with no network.

Reply:
https://acteng.github.io/atip/fail_route_load/scheme.html?authority=Adur
Give it ~10 mins to deploy, then we can play with this. It's set up to grab the wrong route files and fail, so we can see everything that breaks. I added a test (now in `errors.spec.ts` that demonstrates making a new point still works, at least)

Reply:
@dabreegster cannot see how to add anything here.. https://acteng.github.io/atip/fail_route_load/scheme.html?authority=Adur#5.74/52.469/-1.707

Reply:
Ahh. "New point" and freehand polygon work fine. But as soon as you try to click "New route", everything gets stuck. Thanks, we can update the tests accordingly and fix this bug